src/libraries/System.Data.Common/src/System/Data/SQLTypes/SQLGuid.cs

```diff
@@ -16,68 +20,65 @@ namespace System.Data.SqlTypes
     [Serializable]
     [XmlSchemaProvider("GetXsdType")]
     [System.Runtime.CompilerServices.TypeForwardedFrom("System.Data, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
-    public struct SqlGuid : INullable, IComparable, IXmlSerializable, IEquatable<SqlGuid>
+    public struct SqlGuid : INullable, IComparable, ISerializable, IXmlSerializable, IEquatable<SqlGuid>
     {
-        private const int SizeOfGuid = 16;
+        private const int SizeOfGuid = 16; // sizeof(Guid)
 
         // NOTE: If any instance fields change, update SqlTypeWorkarounds type in System.Data.SqlClient.
-        private byte[]? m_value; // the SqlGuid is null if m_value is null
+        private Guid? _value; // the SqlGuid is null if _value is null
 
         // constructor
         // construct a SqlGuid.Null
         private SqlGuid(bool fNull)
         {
-            m_value = null;
+            _value = null;
         }
 
         public SqlGuid(byte[] value)
         {
             if (value == null || value.Length != SizeOfGuid)
                 throw new ArgumentException(SQLResource.InvalidArraySizeMessage);
 
-            m_value = new byte[SizeOfGuid];
-            value.CopyTo(m_value, 0);
-        }
-
-        internal SqlGuid(byte[] value, bool ignored)
-        {
-            if (value == null || value.Length != SizeOfGuid)
-                throw new ArgumentException(SQLResource.InvalidArraySizeMessage);
-
-            m_value = value;
+            _value = new Guid(value);
         }
 
         public SqlGuid(string s)
         {
-            m_value = (new Guid(s)).ToByteArray();
+            _value = new Guid(s);
         }
 
         public SqlGuid(Guid g)
         {
-            m_value = g.ToByteArray();
+            _value = g;
         }
 
         public SqlGuid(int a, short b, short c, byte d, byte e, byte f, byte g, byte h, byte i, byte j, byte k)
             : this(new Guid(a, b, c, d, e, f, g, h, i, j, k))
         {
         }
 
-
-        // INullable
-        public bool IsNull
+        private SqlGuid(SerializationInfo si, StreamingContext sc)
         {
-            get { return (m_value is null); }
+            ArgumentNullException.ThrowIfNull(si);
```

Comment:
The null check isn't needed here. Just let the `NullReferenceException` propagate out.